test/sequential/test-repl-persistent-history.js

```diff
@@ -149,16 +158,44 @@ const tests = [{
   test: [UP],
   expected: [prompt, homedirErr, prompt, replDisabled, prompt]
 }];
+const numtests = tests.length;
+
+
+var testsNotRan = tests.length;
 
+process.on('beforeExit', () =>
+  assert.strictEqual(testsNotRan, 0)
+);
+
+function cleanupTmpFile() {
+  try {
+    // Write over the file, clearing any history
+    fs.writeFileSync(defaultHistoryPath, '');
+  } catch (err) {
+    if (err.code === 'ENOENT') return true;
+  }
+  return true;
+}
 
 // Copy our fixture to the tmp directory
 fs.createReadStream(historyFixturePath)
-  .pipe(fs.createWriteStream(historyPath)).on('unpipe', runTest);
+  .pipe(fs.createWriteStream(historyPath)).on('unpipe', () => runTest());
 
-function runTest() {
+function runTest(assertCleaned) {
   const opts = tests.shift();
   if (!opts) return; // All done
 
+  if (assertCleaned) {
+    try {
+      assert.strictEqual(fs.readFileSync(defaultHistoryPath, 'utf8'), '');
+    } catch (e) {
+      if (e.code !== 'ENOENT') {
+        console.error(`Failed test # ${numtests - tests.length}`);
+        throw e;
+      }
+    }
+  }
```

Comment:
@targos That other bit allows this to run properly.

I forgot to originally comment, but I'm not sure if I should keep this check. it's not strictly necessary and sort-of ends up testing `fs.{write|read}FileSync()` more than anything.